src/components/PopupWithForm.js

```diff
@@ -8,26 +8,37 @@ export default class PopupWithForm extends Popup {
     this._inputs = Array.from(
       this._popupForm.querySelectorAll(".modal__input")
     );
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonText =
+      this._submitButton.childNodes[0].textContent.trim();
   }
 
   _getInputValues() {
     this._inputValues = {};
     this._inputs.forEach((input) => {
       this._inputValues[input.name] = input.value;
     });
-
     return this._inputValues;
   }
 
   setEventListeners() {
     super.setEventListeners();
 
-    this._submitHandler = (evt) => {
+    this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
+      this.renderLoading(true);
+      this._handleFormSubmit(this._getInputValues())
+        .then(() => this.close())
```

Comment:
You duplicate the logics in `index.js` as well. You should decide where to have `then`, `catch`, `finally` and remove redundant promise chains